src/actions/RankTwoTensorIntegralAction.C

```diff
@@ -0,0 +1,86 @@
+#include "AddVariableAction.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "RankTwoTensorIntegralAction.h"
+#include "libmesh/string_to_enum.h"
+
+registerMooseAction("DeerApp", RankTwoTensorIntegralAction,
+                    "add_postprocessor");
+
+InputParameters RankTwoTensorIntegralAction::validParams() {
+  InputParameters params = Action::validParams();
+  params.addClassDescription(
+      "Compute the volume or area integral for each component of the "
+      "provided rank two tensor. Optionally, the compute integrals can be "
+      "scaled using an additional postprocessor (scaling_factor_PP), provided "
+      "as input.");
+  params.addParam<bool>("use_displaced_mesh", true,
```

Comment:
Ah the dreaded option.  As long as you know what you're doing...

Reply:
yes this is why I added an overarching action that does everything for you and sets things correctly. For instance integration for the volumetric strain must always be done on the undeformed mesh as it uses a total Lagrangian formulation that is embedded in the material.